cpp_api/api.cpp

```diff
@@ -0,0 +1,36 @@
+#include <ansi-c/ansi_c_language.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/initialize_goto_model.h>
+
+#include <langapi/mode.h>
+
+#include <util/cmdline.h>
+#include <util/message.h>
+#include <util/options.h>
+
+#include "api.h"
+
+api_depst api_deps;
+extern configt config;
+
+void initialize() {
```

Comment:
I remember that we discussed about introducing a session object to encapsulate all the memory owned by the library so that the client has control about the session/memory lifetime.